plugins/module_utils/ndi.py

```diff
@@ -0,0 +1,452 @@
+        self.compliance_path = "model/aciPolicy/complianceAnalysis"
+        self.epoch_delta_ig_path = "epochDelta/insightsGroup/{0}/fabric/{1}/job/{2}/health/view"
+
+    def get_site_id(self, ig_name, site_name, **kwargs):
+        obj = self.nd.query_obj(self.config_ig_path, **kwargs)
+        for insights_group in obj['value']['data']:
+            if ig_name == insights_group['name']:
+                for site in insights_group['assuranceEntities']:
+                    if site['name'] == site_name:
+                        return site['uuid']
+
+    def get_pre_change_result(self, pcv_results, name, site_id, path,  **kwargs):
+        pcv_result = {}
+        for pcv in pcv_results:
+            if pcv.get("name") == name and pcv.get("fabricUuid") == site_id:
+                pcv_job_id = pcv.get("jobId")
+                pcv_path = '{0}/{1}'.format(path, pcv_job_id)
+                obj = self.nd.query_obj(pcv_path,  **kwargs)
+                pcv_result = obj['value']['data']
+        return pcv_result
+
+    def get_epochs(self, ig_name, site_name):
+        ig_base_path = self.event_insight_group_path.format(ig_name, site_name)
+        path = '{0}/epochs?$size=1&$status=FINISHED&%24epochType=ONLINE%2C+OFFLINE&%24sort=-collectionTime%2C-analysisStartTime'.format(ig_base_path)
+        obj = self.nd.query_obj(path, prefix=self.prefix)
+        return obj['value']['data'][0]
+
+    def query_data(self, path):
+        obj = self.nd.query_obj(path, prefix = self.prefix)
+        return obj['value']['data']
+
+    def query_compliance_score(self, ig_name, site_name, compliance_epoch_id):
+        ig_base_path = self.event_insight_group_path.format(ig_name, site_name)
+        path = "{0}/{1}/complianceScore?%24epochId={2}".format(ig_base_path, self.compliance_path, compliance_epoch_id)
+        return self.query_data(path)
+
+    def query_compliance_count(self, ig_name, site_name, compliance_epoch_id):
+        ig_base_path = self.event_insight_group_path.format(ig_name, site_name)
+        path = "{0}/{1}/count?%24epochId={2}".format(ig_base_path, self.compliance_path, compliance_epoch_id)
+        return self.query_data(path)
+
+    def query_entry(self, ig_name, site_name, epoch_delta_job_id, epoch_choice):
+        epoch_delta_ig_path = self.epoch_delta_ig_path.format(ig_name, site_name, epoch_delta_job_id)
+        path = "{0}/individualTable?%24size=100".format(epoch_delta_ig_path)
+        if epoch_choice:
+            path = "{0}&epochStatus={1}".format(path, epoch_choice)
+        obj = self.nd.query_obj(path, prefix = self.prefix)
+        pages, r = divmod(obj.get("totalItemsCount"), 100)
+        if r > 0 :
+            pages += 1
+        entries = []
+        for page in range(pages):
+            page_path = "{0}&%24page={1}".format(path, page)
+            obj = self.nd.query_obj(page_path, prefix = self.prefix)
+            entries += obj.get('entries')
+        return entries
```

Comment:
Shouldn't you skip the iteration for page = 0 as you already queried it before on line 66?